typecheck.ini

```diff
@@ -0,0 +1,5 @@
+[mypy]
```

Comment:
why are you adding this file we usually just use setup.cfg

Reply:
it would be used by `make typecheck` as commented before. https://github.com/ibis-project/ibis/pull/2161#issuecomment-610424961

Reply:
I am pretty sure that you can have this just pickup setup.cfg

Reply:
The point here is that if we don't have an extra file (in this case typecheck.ini) `make typecheck` will no longer show any errors (because setup.cfg is now ignoring the errors)

So if we want to keep `make typecheck` showing the errors we want to fix in the future, I don't know other way to do it, but I would happy to receive any recommendation.

But it we want that `make typecheck` should just show the errors related to the main configuration (in this case it is ignoring all the errors) there is no need for this new config file.

Reply:
I'm -1 on having 160 mypy errors, and two different systems to check types. One with everything ignored, and another that reports the errors.

I think it makes much more sense to fix the errors in some files or directories, and add validation to those in the CI.